internal/services/containers/kubernetes_cluster_node_pool_resource_test.go

```diff
@@ -565,29 +565,96 @@ func TestAccKubernetesClusterNodePool_upgradeSettings(t *testing.T) {
 
 	data.ResourceTest(t, r, []acceptance.TestStep{
 		{
-			Config: r.upgradeSettingsConfig(data, "2"),
+			Config: r.upgradeSettingsConfig(data, map[string]interface{}{
+				"max_surge":                "2",
+				"drain_timeout_in_minutes": 35,
```

Comment:
Unless there is a good reason to separate testing the update of these properties I would add `node_soak_duration_in_minutes` here and test updating and removing them together. Testing them as individual steps in the same test would lengthen the test duration and drive up the cost/price of it.